rviz_common/include/rviz_common/message_filter_display.hpp

```diff
@@ -59,13 +60,24 @@ class MessageFilterDisplay : public _RosTopicDisplay
   : tf_filter_(nullptr),
     messages_received_(0)
   {
-    // TODO(Martin-Idel-SI): We need a way to extract the MessageType from the template to set a
-    // correct string. Previously was:
-    // QString message_type =
-    //   QString::fromStdString(message_filters::message_traits::datatype<MessageType>());
-    QString message_type = QString::fromStdString("");
-    topic_property_->setMessageType(message_type);
-    topic_property_->setDescription(message_type + " topic to subscribe to.");
+    try {
+      std::string data_type(rosidl_generator_traits::data_type<MessageType>());
```

Comment:
It seems like it would be generally useful to get name directly from the generated code, rather than doing this find-replace logic. I've opened https://github.com/ros2/rosidl/pull/514 with a proposal for such a function.

Reply:
Note, if https://github.com/ros2/rosidl/pull/514 is not approved, then I think re-using the [find-replace implementation rcpputils](https://github.com/ros2/rcpputils/blob/5e9849794d1d242322909bf6746779698cb64ce0/include/rcpputils/find_and_replace.hpp) would make this a bit nicer. I don't know if we need to worry about handling an invalid type name here, since it should be handled when creating the subscription later.

Reply:
Sounds good, thanks for the suggestion and for making ros2/rosidl#514.